File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    l = []
    for i in range(len(progresses)):
        k = (100-progresses[i]) // speeds[i]
        if (100-progresses[i]) % speeds[i] == 0:
            l.append(k)
        else:
            l.append(k+1)
    start = l[0]
    count = 1
    for i in range(1,len(l)):
        if l[i] <= start:
            count +=1
        else:
            answer.append(count)
            count = 1
            start = l[i]
    answer.append(count)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        List<Integer> l = new ArrayList<>();
        for (int i=0; i<progresses.length;i++) {
            if ((100-progresses[i]) % speeds[i] == 0) {
                l.add((100-progresses[i]) / speeds[i]);
            }else {
                l.add((100-progresses[i]) / speeds[i]+1);
            }
        }

        int count = 0;
        int pre_data = l.get(0);
        for (Integer i: l) {
            if (i <= pre_data) {
                count +=1;
            } else {
                answer.add(count);
                count = 1;
                pre_data = i;
            }
        }
        answer.add(count);
        return answer.stream().mapToInt(i->i).toArray();
    }
}